src/algebra/hom/centroid.lean

```diff
@@ -262,16 +262,29 @@ instance : add_comm_group (centroid_hom α) :=
 to_End_injective.add_comm_group _ (by { ext, refl }) (λ _ _, rfl) (λ _, rfl) (λ _ _, rfl)
   (λ _ _, rfl) (λ _ _, by { ext, refl })
 
-instance : ring (centroid_hom α) := to_End_injective.ring _ (by { ext, refl }) (by { ext, refl })
-  (λ _ _, rfl)  (λ _ _, rfl) (congr_fun rfl) (congr_fun₂ rfl) (λ _ _, rfl) (λ _ _, rfl)
-  (λ _ _, by { ext, refl }) (λ n, rfl) (λ n, rfl)
-
 @[simp] lemma coe_neg (f : centroid_hom α) : ⇑(-f) = -f := rfl
 @[simp] lemma coe_sub (f g : centroid_hom α) : ⇑(f - g) = f - g := rfl
 
 @[simp] lemma neg_apply (f : centroid_hom α) (a : α) : (-f) a = - f a := rfl
 @[simp] lemma sub_apply (f g : centroid_hom α) (a : α) : (f - g) a = f a - g a := rfl
 
+@[simp] lemma to_End_zero : (0 : centroid_hom α).to_End = 0 := rfl
+@[simp] lemma to_End_one : (1 : centroid_hom α).to_End = 1 := rfl
+@[simp] lemma to_End_add (x y : centroid_hom α) : (x + y).to_End = x.to_End + y.to_End := rfl
+@[simp] lemma to_End_mul (x y : centroid_hom α) : (x * y).to_End = x.to_End * y.to_End := rfl
+@[simp] lemma to_End_neg (x : centroid_hom α) : (-x).to_End = -x.to_End := rfl
+@[simp] lemma to_End_sub (x y : centroid_hom α) : (x - y).to_End = x.to_End - y.to_End := rfl
+lemma to_End_smul_nat (x : centroid_hom α) (n : ℕ) : (n • x).to_End = n • x.to_End := rfl
+lemma to_End_smul_int (x : centroid_hom α) (n : ℤ) : (n • x).to_End = n • x.to_End := rfl
```

Comment:
```suggestion
lemma to_End_nsmul (x : centroid_hom α) (n : ℕ) : (n • x).to_End = n • x.to_End := rfl
lemma to_End_zsmul (x : centroid_hom α) (n : ℤ) : (n • x).to_End = n • x.to_End := rfl
```

Reply:
We seem to use `smul_nat` and `smul_int` in some names and `nsmul` and `zsmul` in others. It would be good to have a convention and stick to it. CC @eric-wieser

Reply:
The convention is definitely `nsmul`/`zsmul`. There are 228/136 results for `nsmul`/`zsmul` matching full words and less than 50 results for each of `smul_nat`, `nat_smul`, `smul_int`, `int_smul` not matching full words.

Reply:
Shall I do a PR to eliminate the `smul_nat`, `nat_smul`, `smul_int`, `int_smul`?

Reply:
Sure. As I said there are not even that many.

Reply:
Possibly [nat_smul_eq_nsmul](https://github.com/leanprover-community/mathlib/blob/master/src/algebra/module)/basic.lean#L335 should stay as it is? Otherwise  it would be `nsmul_eq_nsmul` which sounds wrong?

Reply:
In any case, I've fixed this PR.